src/prepare_pcap.c

```diff
@@ -101,17 +98,77 @@ size_t get_ethertype_offset(int link, const uint8_t* pktdata)
         /* http://www.tcpdump.org/linktypes/LINKTYPE_LINUX_SLL.html */
         /* pkttype[2], arphrd_type[2], lladdrlen[2], lladdr[8], ethertype[2] */
         offset = 14;
+    } else if (link == DLT_IEEE802_11_RADIO) {
+        /* get RadioTap header length
+        */
+        uint16_t radio_hdr_len = 0;
+        /* http://www.radiotap.org */
+        /* rdtap_version[1], pad[1], rdtap_hdr_len[2], rdtap_flds[4] */
+        memcpy(&radio_hdr_len, pktdata + 2, sizeof(radio_hdr_len));
+#ifdef __HPUX
+        /* http://radiotap.org */
+        /* all data fields in the radiotap header are to be specified in little endian byte-order
+        */
+        radio_hdr_len = htons(radio_hdr_len);
+#endif
```

Comment:
This shouldn't be inside an `ifdef`. Just let htons be there since it is describing the right behaviour. Its super efficiently implemented and gcc/clang should be able to do the right thing.